spark/src/main/java/io/ddf/spark/etl/JoinHandler.java

```diff
@@ -59,26 +62,23 @@ public DDF join(DDF anotherDDF, JoinType joinType, List<String> byColumns, List<
     }
     joinConditionString = joinConditionString.substring(0, joinConditionString.length() - 5); //remove " AND " at the end
 
-    // we will not select column that is already in left table
-    String rightSelectColumns = "";
-    
-    for (String colname : leftColumns) {
-      if (rightColumnNameSet.contains(colname)) {
-        rightColumnNameSet.remove(colname);
-        rightColumnNameSet.add(String.format("%s AS r_%s", colname,colname));
-      }
+    // Add suffix to overlapping columns, use default if not provided
+    if (leftSuffix == null || StringUtils.isEmpty(leftSuffix.trim())) {
+      leftSuffix = "_l";
```

Comment:
SparkSQL/Hive does not accept column names starting with _ though it uses _c0, _c1, ... for anonymous expressions in the SQL query. So I think we should avoid this convention.